containers/src/Data/IntMap/Internal.hs

```diff
@@ -3154,12 +3154,15 @@ fromDistinctAscList = fromAscList
 -- mask must occur consecutively in the list.
 
 fromMonoListWithKey :: (Key -> a -> a -> a) -> [(Key,a)] -> IntMap a
-fromMonoListWithKey f []              = Nil
-fromMonoListWithKey f ((kx,vx) : zs1) = addAll' kx vx zs1
+fromMonoListWithKey f = go
```

Comment:
Note: I went back on an earlier decision and introduced a helper here because I'm slightly worried that otherwise, all the other helpers, which constitute the bulk of the `fromMonoListWithKey` implementation, may be duplicated.

Reply:
Do you consider this ready to merge?